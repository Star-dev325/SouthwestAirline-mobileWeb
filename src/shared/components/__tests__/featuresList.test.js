import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import React from 'react';
import FeaturesList from 'src/shared/components/featuresList';

describe('FeaturesList', () => {
  it('should render correct styles without icon ', () => {
    const { container } = createComponent([
      {
        icon: 'wrong',
        label: 'The Feature'
      }
    ]);

    expect(container).toMatchSnapshot();
  });

  it('should render correct styles with icon ', () => {
    const { container } = createComponent([
      {
        icon: 'check',
        label: 'The Feature'
      }
    ]);

    expect(container).toMatchSnapshot();
  });

  it('should render without suffix', () => {
    const { container } = createComponent([
      {
        icon: 'circle',
        label: 'The Feature'
      }
    ]);

    expect(container.querySelector('.icon_bullet')).not.toBeNull();
    expect(container.querySelector('.features-list--text').textContent).toEqual('The Feature');
  });

  it('should render with suffix', () => {
    const { container } = createComponent([
      {
        icon: 'plus',
        label: 'The Feature',
        suffix: '2'
      }
    ]);

    expect(container.querySelector('.icon_plus')).not.toBeNull();
    expect(container.querySelector('.features-list--text').textContent).toEqual('The Feature2');
  });
});

const createComponent = (features) => render(<FeaturesList {...{ features }} />);
