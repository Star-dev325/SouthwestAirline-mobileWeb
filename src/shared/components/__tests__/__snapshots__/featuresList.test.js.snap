// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturesList should render correct styles with icon  1`] = `
<div>
  <ul
    class="features-list"
  >
    <li
      class="icon icon_check features-list--item"
    >
      <span
        class="features-list--text"
      >
        The Feature
      </span>
    </li>
  </ul>
</div>
`;

exports[`FeaturesList should render correct styles without icon  1`] = `
<div>
  <ul
    class="features-list"
  >
    <li
      class="icon icon_undefined features-list--no-icon"
    >
      <span
        class="features-list--text"
      >
        The Feature
      </span>
    </li>
  </ul>
</div>
`;
